influxql/query_manager.go

```diff
@@ -49,8 +56,33 @@ type QueryParams struct {
 	// Not required, but highly recommended. If this channel is not set, the
 	// query needs to be manually managed by the caller.
 	InterruptCh <-chan struct{}
+
+	// Holds any error thrown by the QueryManager if there is a problem while
+	// executing the query. Optional.
+	Error *QueryError
+}
+
+// QueryError is an error thrown by the QueryManager when there is a problem
+// while executing the query.
+type QueryError struct {
+	err error
+	mu  sync.Mutex
 }
```

Comment:
Why is the mutex on the `QueryError` and not on the `QueryTask` when you're setting the error?